grouper/models/public_key.py

```diff
@@ -2,6 +2,63 @@
 from sqlalchemy import Column, Integer, String, Text, ForeignKey, DateTime
 from sqlalchemy.orm import relationship
 from grouper.models.base.model_base import Model
+from grouper.models.public_key_tag_map import PublicKeyTagMap
+
+
+def _perm_list_to_dict(perms):
+    # type: List[Permission] -> Dict[str, Dict[str, Permission]]
+    """Converts a list of Permission objects into a dictionary indexed by the permission names.
+    That dictionary in turn holds another dictionary which is indexed by permission argument, and
+    stores the Permission object
+
+    Args:
+        perms: a list containing Permission objects
+
+    Returns:
+        a dictionary with the permission names as keys, and has as values another dictionary
+        with permission arguments as keys and Permission objects as values
+    """
+    ret = dict()
```

Comment:
nit: prefer using the literal, i.e. `ret = {}`

better yet, use `defaultdict` and avoid line 24.